suite-native/device/src/hooks/useRenderDeviceDangerBanner.ts

```diff
@@ -24,7 +24,8 @@ import { selectIsOnboardingFinished } from '@suite-native/settings';
 
 export const useRenderDeviceDangerBanner = () => {
     const setBannerVariant = useSetAtom(deviceDangerBannerAtom);
-    const lastRoute = useNavigationState(state => state?.routes.at(-1)?.name);
+    const navigation = useNavigation();
+    const lastRoute = navigation.getState()?.routes.at(-1)?.name;
```

Comment:
From what I understand, `getState` is mostly useful for reading a static value, but if the last route changes it won’t trigger a re-render. Therefore, if `useRenderDeviceDangerBanner` is mounted all the time, the value won’t update. https://reactnavigation.org/docs/use-navigation-state?config=dynamic